### PR TITLE
ENH: Support nearest neighbor interpolator for montaging label maps

### PR DESCRIPTION
**Pull request by dzenanz**
See https://discourse.itk.org/t/how-to-use-itkmontage-to-stitch-segmentation-label-images/6611.

### CODE REVIEW

File: wrapping/itkTileMergeImageFilter.wrap [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
-  foreach(d ${ITK_WRAP_IMAGE_DIMS})
-    foreach(t ${WRAP_ITK_SCALAR})
-      itk_wrap_template("${ITKM_I${t}${d}}" "${ITKT_I${t}${d}}, double")
-    endforeach()
-    foreach(t ${WRAP_ITK_RGB})
-      itk_wrap_template("${ITKM_I${t}${d}}" "${ITKT_I${t}${d}}, itk::RGBPixel<float>")
-    endforeach()
-    foreach(t ${WRAP_ITK_RGBA})
-      itk_wrap_template("${ITKM_I${t}${d}}" "${ITKT_I${t}${d}}, itk::RGBAPixel<float>")
+  foreach(iif "LinearInterpolateImageFunction;NearestNeighborInterpolateImageFunction")
```
Comment (thewtex) [inline wrapping/itkTileMergeImageFilter.wrap:4]:
```suggestion
  foreach(iif LinearInterpolateImageFunction NearestNeighborInterpolateImageFunction)
```

File: wrapping/itkTileMergeImageFilter.wrap [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
-      itk_wrap_template("${ITKM_I${t}${d}}" "${ITKT_I${t}${d}}, itk::RGBAPixel<float>")
+  foreach(iif LinearInterpolateImageFunction NearestNeighborInterpolateImageFunction)
+    if (${iif} STREQUAL "LinearInterpolateImageFunction")
+      set(abbr "L")  # abbreviation to use in Python wrapping type name
+    else()
+      set(abbr "NN")
+    endif()
+    foreach(d ${ITK_WRAP_IMAGE_DIMS})
+      foreach(t ${WRAP_ITK_SCALAR})
+        itk_wrap_template("${ITKM_I${t}${d}}${abbr}" "${ITKT_I${t}${d}}, double, ${iif}")
```
Comment (thewtex) [inline wrapping/itkTileMergeImageFilter.wrap:12]: itk_wrap_template("${ITKM_I${t}${d}}${abbr}" "${ITKT_I${t}${d}}, double, itk::${iif}")
```suggestion
        itk_wrap_template("${ITKM_I${t}${d}}${abbr}" "${ITKT_I${t}${d}}, double, itk::${iif}")
```

File: wrapping/itkTileMergeImageFilter.wrap [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
+      endforeach()
+      foreach(t ${WRAP_ITK_RGB})
+        itk_wrap_template("${ITKM_I${t}${d}}${abbr}" "${ITKT_I${t}${d}}, itk::RGBPixel<float>, ${iif}")
```
Comment (thewtex) [inline wrapping/itkTileMergeImageFilter.wrap:15]:
```suggestion
        itk_wrap_template("${ITKM_I${t}${d}}${abbr}" "${ITKT_I${t}${d}}, itk::RGBPixel<float>, itk::${iif}")
```

File: wrapping/itkTileMergeImageFilter.wrap [hunk 4]
```diff
@@ -1,13 +1,22 @@
...
+      endforeach()
+      foreach(t ${WRAP_ITK_RGBA})
+        itk_wrap_template("${ITKM_I${t}${d}}${abbr}" "${ITKT_I${t}${d}}, itk::RGBAPixel<float>, ${iif}")
```
Comment (thewtex) [inline wrapping/itkTileMergeImageFilter.wrap:18]:
```suggestion
        itk_wrap_template("${ITKM_I${t}${d}}${abbr}" "${ITKT_I${t}${d}}, itk::RGBAPixel<float>, itk::${iif}")
```

**Comment by dzenanz** [comment]
This is expected to pass after rebasing on #230.

**Review by thewtex — APPROVED** [review]

**Pull request merged**